parcel-resolver-veda/index.d.ts

```diff
@@ -62,13 +62,27 @@ declare module 'veda' {
   // resolved from DatasetLayerCompareSTAC or DatasetLayerCompareInternal. The
   // difference with a "base" dataset layer is not having a name and
   // description.
-  export interface DatasetLayerCompareNormalized
+  export interface DatasetLayerCompareBase
     extends DatasetLayerCommonCompareProps {
     id: string;
     stacCol: string;
     type: DatasetLayerType;
   }
 
+  interface DatasetLayerCompareNoLegend extends DatasetLayerCompareBase {
+    legend?: never;
+    name?: never;
+    description?: never;
+  }
+      
+  interface DatasetLayerCompareWLegend extends DatasetLayerCompareBase {
+    legend: LayerLegendCategorical | LayerLegendGradient | undefined;
```

Comment:
This passes ts check but does not real accurately reflect how typ should work. Because if the legend is undefined  - name and description should not exist. I tried several things but nothing seemed to work. Do you @danielfdsilva @nerik have an idea how I can type this better?